include/openthread/border_router.h

```diff
@@ -52,6 +52,20 @@ extern "C" {
  *
  */
 
+/**
+ * This method initializes the Border Router on given infrastructure interface.
+ *
+ * @param[in]  aInstance      A pointer to an OpenThread instance.
+ * @param[in]  aInfraIfIndex  The infrastructure interface index.
+ * @param[in]  aInfraIfName   The infrastructure interface name.
+ *
+ * @retval  OT_ERROR_NONE          Successfully started the Border Router on given infrastructure.
+ * @retval  OT_ERROR_INVALID_ARGS  Failed to start the Border Router.
+ * @retval  OT_ERROR_ALREADY       The Border Router has already been initialized.
+ *
+ */
+otError otBorderRouterInit(otInstance *aInstance, uint32_t aInfraIfIndex, const char *aInfraIfName);
```

Comment:
Agree. But prefer `otBorderRouterSetInfraIf` since we currently supports only a single infra if. We can rename to `...Add...` when we supported multiple infra if.

Reply:
`otBorderRouterInit` is misleading because it seems to initialize the `Thread Border Router` functionality by name, however it actually only initialize the Border Routing Infrastructure network interface manager.

Reply:
It also loads the OMR & on-link prefix from settings (and/or generate random prefixes if there are none in settings).

Reply:
fine to me.